packages/react-button/src/components/Button/Button.tsx

```diff
@@ -1,9 +1,27 @@
-import { compose, createClassResolver } from '@fluentui/react-compose';
-import { ButtonBase } from './ButtonBase';
+import * as React from 'react';
+import { createButton } from './createButton';
 import { ButtonProps } from './Button.types';
+import { useFocusRects } from '@uifabric/utilities';
+import { makeClasses } from '@fluentui/react-compose/lib/next/index';
+import { useInlineTokens } from '@fluentui/react-theme-provider';
 import * as classes from './Button.scss';
 
-export const Button = compose<'button', ButtonProps, ButtonProps, {}, {}>(ButtonBase, {
-  classes: createClassResolver(classes),
-  displayName: 'Button',
+// Create a hook to resolve classnames.
+export const useButtonClasses = makeClasses(classes);
+
+/**
+ * Define a styled Button, using the `createButton` factory.
+ */
+export const Button = React.forwardRef<HTMLElement, ButtonProps>((props, ref) => {
+  const { render, state } = createButton(props, ref);
+
+  // Apply styling.
+  useButtonClasses(state);
+  useFocusRects(state.ref);
```

Comment:
It looks that `state` returns `any` or something close to it.

- `ref` that comes from `forwardRef` defaults to `null` ⏭️ 
- after merging via `mergeProps` `ref` will be still `null` ⏭️ 
- `useFocusRects` will point to invalid `window`

https://github.com/microsoft/fluentui/issues/14062